src/SymfonyCache/KernelClient.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\SymfonyCache;
+
+use Http\Client\HttpAsyncClient;
+use Http\Promise\FulfilledPromise;
+use Psr\Http\Message\RequestInterface;
+use Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory;
+use Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Zend\Diactoros\ServerRequest;
+
+/**
+ * An implementation of HttpAsyncClient that allows direct routing through the
+ * Symfony HttpCache kernel without executing a real HTTP request.
+ * It uses the HttpFoundationFactory and the Zend DiactorosFactory to convert
+ * between PSR-7 requests and responses. Both are optional dependencies of this
+ * package thus existence of the respective classes is checked for in the
+ * constructor of this client. It is only needed if you have a single node
+ * setup of Symfony and serves as kind of a shortcut for easier configuration.
+ * If you use Varnish or have a multiple node Symfony setup, this class is entirely
+ * useless to you and you can happily ignore it.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class KernelClient implements HttpAsyncClient
```

Comment:
could we instead of having a "mock" http client, create a Dispatcher interface that the HttpDispatcher implements, and that this class would implement? and then refactor a base class out of HttpDispatcher for the validatation and queuing mechanism, but with a different `flush`.